### PR TITLE
Upgrade to latest

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

- Collection maintanence (upgrade to latest GH API specs)
- Convert credentials to blocks
- Remove scripts/update.py so scripts/generate.py is the sole source of truth / regeneration
- Remove _edge tasks

Supporting PRs
https://github.com/PrefectHQ/prefect-collection-generator/pull/26
https://github.com/PrefectHQ/prefect-collection-generator/pull/25

## Checklist
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-github/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_github/credentials.py [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+        Load stored GitHub credentials:
+        ```python
+        from prefect_github import GitHubCredentials
+        github_credentials_block = GitHubCredentials.load("BLOCK_NAME")
+        ```
     """
 
-    token: str
+    _block_type_name = "GitHub Credentials"
+    # _logo_url = "<LOGO_URL_HERE>"
```
Comment (ahuang11) [inline prefect_github/credentials.py:24]:
```suggestion
    # _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/187oCWsD18m5yooahq1vU0/ace41e99ab6dc40c53e5584365a33821/github.png?h=250"  # noqa
```

File: prefect_github/credentials.py [hunk 2]
```diff
@@ -32,14 +41,18 @@ def get_endpoint(self) -> HTTPEndpoint:
             @flow
             def example_get_endpoint_flow():
                 token = "consumer_key"
-                github_credentials = GitHubCredentials(token)
+                github_credentials = GitHubCredentials(token=token)
```
Comment (desertaxle) [inline prefect_github/credentials.py:44]: I don't see `token` declared above this line.
```suggestion
                github_credentials = GitHubCredentials(token="ghu_xxxxxxxx")
```

File: prefect_github/mutations.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 """
-This is a module for interacting with GitHub Mutation  tasks.
+This is a module for interacting with GitHub  tasks.
```
Comment (desertaxle) [inline prefect_github/mutations.py:2]: wow, such nit, so small
```suggestion
This is a module for interacting with GitHub tasks.
```

File: scripts/generate.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#     checkout="generated_graphql",
+#     extra_context=extra_context,
+#     overwrite_if_exists=True
+# )
+# REPO_DIRECTORY = THIS_DIRECTORY / "prefect_github"  # redirects repo_directory
+
+# UPDATE THESE AS DESIRED
+
+service_name = "GitHub"  # e.g. GitHub
+service_url = "https://api.github.com/graphql"  # e.g. https://api.github.com/graphql
```
Comment (desertaxle) [inline scripts/generate.py:34]: Probably don't need these comments
```suggestion
service_name = "GitHub"
service_url = "https://api.github.com/graphql"
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Just a few small requests

File: prefect_github/credentials.py (hunk 2 above)
Comment (ahuang11) [reply prefect_github/credentials.py:44]: token is declared I think, but I updated the generator to replace the "consumer_key" string

**Review by desertaxle — APPROVED** [review]

**Pull request merged**